src/translator/html.cpp

```diff
@@ -494,11 +504,10 @@ HTML::HTML(std::string &&source, bool process_markup) {
       } break;
 
       case markup::Scanner::TT_TAG_START:
-        // If it makes sense to treat this element as a break in a word (e.g.
-        // <br>, <img>, <li>) make sure it does so in this text as well.
+        // Treat HTML tags always as spaces that break up words.
         // TODO: Strong assumption here that the language uses spaces to
         // separate words
-        if (isBlockElement(scanner.tag()) && !source.empty() && source.back() != ' ') source.push_back(' ');
+        if (!source.empty() && source.back() != ' ') source.push_back(' ');
```

Comment:
```
$ cat | bergamot translate -m en-es-tiny --html=1 --alignment=1 2> /dev/null
<a href="Tintin">Tintin</a>'s dog is a good dog.
El perro de <a href="Tintin">Tintín</a> es un buen perro.

$ cat | bergamot translate -m en-es-tiny --html=1 --alignment=1 2> /dev/null
<a href="Tintin">Tintin</a> 's dog is a good dog.
<a href="Tintin">El</a> perro <a href="Tintin">de Tintín</a> es un buen perro.
```

Reply:
it seems to be unrelated. Something about the model thinking the `'` aligning best with `El`.

Without space:
<img width="1072" alt="without-space" src="https://user-images.githubusercontent.com/198639/148358549-a4af4624-84e3-4777-a2ed-e8c40aa59d9a.png">

With space: (sorry it's very hard to see but here you can see a bit of space between `</a>` and `'` in the top row)
<img width="1072" alt="with-space" src="https://user-images.githubusercontent.com/198639/148358529-1a5d912a-eb07-48c8-8d4a-b59985a1a126.png">

Reply:
@jelmervdl Is there a source that generates the images above? I'm interested in saving some time at integrating it into an HTML rendering of sorts if it's in python or even C++ something. Thank you for providing the visualization, I have opened #298 about this which I believe could improve the library.

To be honest, I was just engineering a valid counter-example where modifying HTML will have an effect (the comment is not about the exact line if you're looking at it this way). I can also see benefits to this PR's line of thinking. The following fragment used to work well when `tidy` indented the divs (although I think the solution should be fixing the non-semantic HTML by the author). 

```
<div class="person"><a href="https://jerinphilip.github.io/"><img src="people/jerin.jpg" height="150" width="150" alt="Jerin Philip"/><div>Jerin Philip</div></a>Research Assistant</div>
```

`PhilipResearch Assistant` affects translation, once HTML is stripped.

Perhaps we should try selective dropping of spaces (`<div>`, `<sup>` etc) with `libtidy` as a reference which interrupts the flow and is guaranteed to be unaffected by spaces. But I guess whatever `libtidy` is doing as this is based on tags and not requiring stats NLP to resolve ambiguity can and should be the responsibility of the web browser.

Reply:
That's what the original implementation did: Everything not `{"abbr", "a", "b", "em", "i", "kbd", "mark", "math", "output", "q", "ruby", "small", "span", "strong", "sub", "sup", "time", "u", "var", "wbr", "ins", "del"}` added whitespace (if it wasn't already there) but those tags didn't.

Note that there was a bug where a closing tag does not add whitespace, regardless of the element.